### ISSUE TITLE
FreeBSD blackhole bug

### ISSUE THREAD

**Issue by tpynak** [nicolargo/glances#1202]
Hi all.
Using Glances on FreeBSD but during optimizing my server I found a problem with glances and system option like blackhole.

So, glances works well, until I will execute commands (or add to sysctl) the following:
sysctl -w net.inet.tcp.blackhole=2
sysctl -w net.inet.udp.blackhole=1

After doing it, glances just loading, and nothing more happens. Until I press ctrl+c. I waited for about 5 minutes.
If I switch those variables back to 0 (blackholes), it again works well.

Versions:
glances --version
Glances v2.11.1 with psutil v5.4.0
python --version
Python 2.7.13

FreeBSD xxx.xxx.com 10.3-RELEASE-p20 FreeBSD 10.3-RELEASE-p20 #0: Wed Jul 12 03:13:07 UTC 2017     root@amd64-builder.daemonology.net:/usr/obj/usr/src/sys/GENERIC  amd64

**Comment by nicolargo** [comment]
Strange behavior... Can you disable all the optional plugins and test again:

    glances --disable-alert --disable-amps --disable-cloud --disable-diskio --disable-docker --disable-folders --disable-fs --disable-gpu --disable-hddtemp --disable-ip  --disable-network --disable-now --disable-ports --disable-process --disable-raid --disable-sensors --disable-wifi --disable-history

**Comment by tpynak** [comment]
Hi, I have tried, and it works. Than I started to disable one by one, and found that it works well with: --disable-sensors, without it glances does not loading.

**Comment by nicolargo** [comment]
@tpynak Can you test the following command lines:

```
python
>>> import psutil
>>> psutil.sensors_temperatures()
>>> psutil.sensors_fans()
>>> psutil.sensors_battery()
```

Thanks !

**Comment by tpynak** [comment]
root@blogs2:~ # python
Python 2.7.13 (default, Aug 15 2017, 01:13:21)
[GCC 4.2.1 Compatible FreeBSD Clang 3.4.1 (tags/RELEASE_34/dot1-final 208032)] on freebsd10
Type "help", "copyright", "credits" or "license" for more information.
>>> import psutil
>>> psutil.sensors_temperatures()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'module' object has no attribute 'sensors_temperatures'
>>> psutil.sensors_fans()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'module' object has no attribute 'sensors_fans'
>>> psutil.sensors_battery()
>>>




Output is the same with blackhole on/off

**Comment by nicolargo** [comment]
Ok look like a PSUtil issue. Just open it: https://github.com/giampaolo/psutil/issues/1195

**Comment by tpynak** [comment]
Yes, probably. But the strangest thing is - blackhole setting. Glances loads well (without disabling third-party plugins) if blackhole turned off. But if it turns on, as I stated before - glances is not loading, just getting stuck. How it is connected?

**Comment by nicolargo** [comment]
As analyzed by @giampaolo PsUtilis only compatible with sensors_battery() on FreeBSD OS.

Had to limit the sensor plugin to Linux and Windows OS.

**Comment by tpynak** [comment]
Thanks for your reply, so - waiting for an update than?

**Comment by nicolargo** [comment]
Yes. Will be corrected in Glances 3.0. As a workaround you can use the
--disable-sensors option on the command line.
